playwright/_impl/_api_structures.py

```diff
@@ -64,9 +64,10 @@ class Geolocation(TypedDict, total=False):
     accuracy: Optional[float]
 
 
-class HttpCredentials(TypedDict):
+class HttpCredentials(TypedDict, total=False):
```

Comment:
Why total=false, isn't it enough that origin is optional?

Reply:
total means that all the properties need to be specified, by default all need to be specified, Optional doesn't help there unfortunately, we used that in the past for it too.